tests/routing.test.ts

```diff
@@ -0,0 +1,134 @@
+import { describe, test, expect, beforeAll, afterAll } from "bun:test"
+import { exec } from "node:child_process"
+import { promisify } from "node:util"
+import { mkdirSync, existsSync } from "node:fs"
+import {
+  convertCircuitJsonToDsnString,
+  mergeDsnSessionIntoDsnPcb,
+  parseDsnToDsnJson,
+  type DsnPcb,
+  type DsnSession,
+  convertDsnPcbToCircuitJson,
+} from "dsn-converter"
+
+const execAsync = promisify(exec)
+let containerId: string
+
+describe("freerouting routing", () => {
+  beforeAll(async () => {
+    if (!existsSync("tests/tests-data")) {
+      mkdirSync("tests/tests-data", { recursive: true })
+    }
+
+    const { stdout } = await execAsync(
+      "docker run -d -p 37864:37864 ghcr.io/tscircuit/freerouting:master",
+    )
+    containerId = stdout.trim()
+
+    await new Promise((resolve) => setTimeout(resolve, 2000))
+
+    await execAsync("bun ./cli.ts config set-api-url http://localhost:37864")
+    await execAsync(
+      "bun ./cli.ts config set-profile e9866fac-e7ae-4f9f-a616-24ec577aa461",
+    )
+
+    try {
+      const { stdout: healthCheck } = await execAsync(
+        "bun ./cli.ts system status",
+      )
+      console.log("Server health check:", healthCheck)
+    } catch (error) {
+      console.error("Server health check failed:", error)
+      throw error
+    }
+  })
+
+  afterAll(async () => {
+    await execAsync(`docker stop ${containerId}`)
+    await execAsync(`docker rm ${containerId}`)
+
+    await execAsync("bun ./cli.ts config reset")
+  })
+
+  test("should route a simple circuit", async () => {
+    try {
+      const { stdout: sessionOutput } = await execAsync(
+        "bun ./cli.ts session create",
+      )
+      expect(sessionOutput).toContain("id:")
+
+      const { stdout: jobOutput } = await execAsync(
+        "bun ./cli.ts job create --name test-routing",
+      )
+      expect(jobOutput).toContain("id:")
+
+      const circuitJson = await Bun.file(
+        "tests/tests-data/circuit-json-with-route.json",
```

Comment:
this is the convention for test-data

```suggestion
        "tests/assets/circuit-json-with-route.json",
```